pkg/chain/ethereum/tbtc.go

```diff
@@ -370,214 +674,195 @@ func (tc *TbtcChain) GetDKGState() (tbtc.DKGState, error) {
-) subscription.EventSubscription {
-	ctx, cancelCtx := context.WithCancel(context.Background())
-	blocksChan := mwr.blockCounter.WatchBlocks(ctx)
+	uint256Type, err := abi.NewType("uint256", "uint256", nil)
+	if err != nil {
+		return dkg.ResultHash{}, err
+	}
+	bytesType, err := abi.NewType("bytes", "bytes", nil)
+	if err != nil {
+		return dkg.ResultHash{}, err
+	}
+	uint8SliceType, err := abi.NewType("uint8[]", "uint8[]", nil)
+	if err != nil {
+		return dkg.ResultHash{}, err
+	}
 
-	go func() {
-		for {
-			select {
-			case block := <-blocksChan:
-				// Generate an event every 1000th block starting from block 250.
-				// The shift is done in order to avoid overlapping with beacon
-				// DKG test loop.
-				shift := uint64(250)
-				if block >= shift && (block-shift)%1000 == 0 {
-					// The seed is keccak256(block).
-					blockBytes := make([]byte, 8)
-					binary.BigEndian.PutUint64(blockBytes, block)
-					seedBytes := crypto.Keccak256(blockBytes)
-					seed := new(big.Int).SetBytes(seedBytes)
-
-					mwr.currentDkgStartBlock = big.NewInt(int64(block))
-
-					go handler(&tbtc.DKGStartedEvent{
-						Seed:        seed,
-						BlockNumber: block,
-					})
-				}
-			case <-ctx.Done():
-				return
-			}
-		}
-	}()
+	bytes, err := abi.Arguments{
+		{Type: uint256Type},
+		{Type: bytesType},
+		{Type: uint8SliceType},
+		{Type: uint256Type},
+	}.Pack(
+		chainID,
+		groupPublicKey,
+		misbehavedMembersIndexes,
+		startBlock,
+	)
+	if err != nil {
+		return dkg.ResultHash{}, err
+	}
 
-	return subscription.NewEventSubscription(func() {
-		cancelCtx()
-	})
+	return dkg.ResultHash(crypto.Keccak256Hash(bytes)), nil
 }
 
-func (mwr *mockWalletRegistry) OnDKGResultSubmitted(
-	handler func(event *tbtc.DKGResultSubmittedEvent),
-) subscription.EventSubscription {
-	mwr.dkgResultSubmissionHandlersMutex.Lock()
-	defer mwr.dkgResultSubmissionHandlersMutex.Unlock()
-
-	// #nosec G404 (insecure random number source (rand))
-	// Temporary test implementation doesn't require secure randomness.
-	handlerID := rand.Int()
+func (tc *TbtcChain) IsDKGResultValid(
+	dkgResult *tbtc.DKGChainResult,
+) (bool, error) {
+	outcome, err := tc.walletRegistry.IsDkgResultValid(
+		convertDkgResultToChainFormat(dkgResult),
+	)
+	if err != nil {
+		return false, fmt.Errorf("cannot check result validity: [%v]", err)
+	}
 
-	mwr.dkgResultSubmissionHandlers[handlerID] = handler
+	return parseDkgResultValidationOutcome(&outcome)
+}
 
-	return subscription.NewEventSubscription(func() {
-		mwr.dkgResultSubmissionHandlersMutex.Lock()
-		defer mwr.dkgResultSubmissionHandlersMutex.Unlock()
+// parseDkgResultValidationOutcome parses the DKG validation outcome and returns
+// a boolean indicating whether the result is valid or not. The outcome parameter
+// must be a pointer to a struct containing a boolean flag as the first field.
+//
+// TODO: Find a better way to get the validity flag. This would require
+//       changes in the contracts binding generator.
```

Comment:
Can you please open an issue in `keep-common` and link it from here?

Reply:
Opened https://github.com/keep-network/keep-common/issues/111